catalog/resource_lakehouse_monitor.go

```diff
@@ -0,0 +1,120 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, DefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.LakehouseMonitors.GetByFullName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusActive {
+			return nil
+		}
+		if endpoint.Status == catalog.MonitorInfoStatusMonitorStatusError || endpoint.Status == catalog.MonitorInfoStatusMonitorStatusFailed {
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", endpoint.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+}
+
+func ResourceLakehouseMonitor() common.Resource {
+	monitorSchema := common.StructToSchema(
+		catalog.MonitorInfo{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.CustomizeSchemaPath(m, "assets_dir").SetRequired()
+			common.CustomizeSchemaPath(m, "output_schema_name").SetRequired()
+			common.CustomizeSchemaPath(m, "table_name").SetRequired()
```

Comment:
This does not seem to match any field in the create request. What field does this correspond to? I think this should be `SetReadOnly()`.

Reply:
We are using `MonitorInfo` as the base struct and this is part of that struct

Reply:
Alright. Sounds reasonable to me.